modules/apps/remote-app/remote-app-client-js/src/main/resources/META-INF/resources/js/index.ts

```diff
@@ -323,6 +323,17 @@ function serializeBody(body: BodyInit | null): StructuredClonable {
 	} else {
 		// warn about unserializable body type (eg. FormData, search params,
 		// readable stream),
+		if (body instanceof FormData) {
+			const formData: any = {};
```

Comment:
We should try really hard to avoid `any` in the type system, because it punches a massive whole in the safety net that it provides.

Here we could go with something like:

```js
const formData: {[key: string]: File | string} = {};
```

Reply:
Thanks I actually used `any` because I didn't know what else to do #bashguycomment

Reply:
The basic approach for these things is to:

1. Ask yourself what possible types the value could have.
2. If you don't know, look at [the TypeScript types](https://github.com/microsoft/TypeScript/blob/2428ade1a91248e847f3e1561e31a9426650efee/lib/lib.dom.d.ts#L19942).
3. Write down the types.
4. If the compiler complains, study the error message because it probably tells you want you need to do next.
5. Repeat.